pkg/ebpf/tracer.go

```diff
@@ -644,16 +637,83 @@ func (t *Tracer) populatePortMapping(mp *bpflib.Map) ([]uint16, error) {
 	return closedPortBindings, nil
 }
 
-func (t *Tracer) determineConnectionDirection(conn *ConnectionStats) ConnectionDirection {
-	if conn.Type == UDP {
-		return NONE
+func (t *Tracer) setConnectionDirections(connections []ConnectionStats) {
+
+	type connKey struct {
+		Address util.Address
+		Port    uint16
+		Type    ConnectionType
+	}
+
+	newConnKey := func(connStat *ConnectionStats, RAddr bool) connKey {
+		key := connKey{}
+		key.Type = connStat.Type
+		// RAddr indicates if we're making a key out of the LAddr or RAddr
+		// of a given connection
+		if RAddr {
+			if connStat.IPTranslation == nil {
+				key.Address = connStat.Dest
+				key.Port = connStat.DPort
+			} else {
+				key.Address = connStat.IPTranslation.ReplSrcIP
+				key.Port = connStat.IPTranslation.ReplSrcPort
+			}
+		} else {
+			if connStat.IPTranslation == nil {
+				key.Address = connStat.Source
+				key.Port = connStat.SPort
+			} else {
+				key.Address = connStat.IPTranslation.ReplDstIP
+				key.Port = connStat.IPTranslation.ReplDstPort
+			}
+		}
+		return key
 	}
 
-	if t.portMapping.IsListening(conn.SPort) {
-		return INCOMING
+	connLAddrMap := make(map[connKey]struct{}, 0)
+	keyWithLAddr := connKey{}
```

Comment:
I don't think this variable needs to be declared outside the for loop immediately below this line.